runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/operation/SdkHttpOperation.kt

```diff
@@ -59,6 +63,7 @@ public class SdkHttpOperation<I, O>(
 /**
  * Round trip an operation using the given [HttpHandler]
  */
+context(TraceSpan)
```

Comment:
I'm guessing this is the most contentious decision. What I'm wondering is how this actually plays out. It seems it will propagate `context(TraceSpan)` all over the place. Of course passing it down explicitly is roughly the same result. 

The alternative (which I'm not saying is better just thinking out loud) is to thread it into `ExecutionContext` which is already [a placeholder for per operation metadata](https://github.com/awslabs/smithy-kotlin/blob/cfa0fd3a30b4c50b75485786f043d4e2ad803f55/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/client/ExecutionContext.kt#L9-L12). 

I think the way to think about which route may be better is to answer the question "how deep are nested child spans going to be"? If *most* child spans are created off of the operation specific span then `ExecutionContext` becomes more attractive since we already thread it through so many places (e.g. every middleware, http engine, etc). If we have mostly deeply nested trace spans then creating child spans off the operation span doesn't help and putting into execution context makes less sense.

I'd like to explore further why we think context receivers is the right choice and what the difference between approaches looks like (which is sort of hard to do in this PR since we create no child spans and can't see any output yet)